### PR TITLE
Fix LoadStreams iterator

### PR DESCRIPTION
**Pull request by rnoriega**
Eliminates repeated iterations over LoadStreams, waiting for at least one new image and returning only these #3559.

I have read the CLA Document and I sign the CLA

### Summary
🔄🚫🚀

Improved stream loading logic in `stream_loaders.py`. Added a loop to wait for images, reset and filter attributes, and avoid empty or None values.

> _`LoadStreams` waits_
> _for images to appear_
> _then resets `self.imgs`_

### Walkthrough
*  Wait for at least one image from streams and filter sources and im0 lists ([link](https://github.com/ultralytics/ultralytics/pull/3584/files?diff=unified&w=0#diff-be9cf5c74e77e0cde8d630f233c18b23147891ede552e737c1ad702c08f7b652L101-R115))
*  Reset self.imgs to None after copying available images ([link](https://github.com/ultralytics/ultralytics/pull/3584/files?diff=unified&w=0#diff-be9cf5c74e77e0cde8d630f233c18b23147891ede552e737c1ad702c08f7b652L101-R115))

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR appears to be a placeholder or mistake, as there are no code changes.

### 📊 Key Changes
- None. The PR contains no actual code or documentation changes.

### 🎯 Purpose & Impact
- Since there are no changes, there is no direct impact on users or the codebase.
- It may have been opened by accident or used for discussion purposes only.

### CODE REVIEW

**Comment by rnoriega** [comment]
I have read the CLA Document and I sign the CLA

**Comment by glenn-jocher** [comment]
@rnoriega what about just inserting a delay corresponding to the FPS into the code. The current delay is 0.0, so we could update this like this (last line in update() method):

```python
    def update(self, i, cap, stream):
        """Read stream `i` frames in daemon thread."""
        n, f = 0, self.frames[i]  # frame number, frame array
        while cap.isOpened() and n < f:
            n += 1
            cap.grab()  # .read() = .grab() followed by .retrieve()
            if n % self.vid_stride == 0:
                success, im = cap.retrieve()
                if success:
                    self.imgs[i] = im
                else:
                    LOGGER.warning('WARNING ⚠️ Video stream unresponsive, please check your IP camera connection.')
                    self.imgs[i] = np.zeros_like(self.imgs[i])
                    cap.open(stream)  # re-open stream if signal was lost
            if self.fps[i]:
                time.sleep(1 / self.fps[i])  # wait time adjusted to match video frame rate
```

File: ultralytics/yolo/data/dataloaders/stream_loaders.py [hunk 1]
```diff
@@ -98,8 +98,20 @@ def __next__(self):
...
+        # Wait for an image to be available
+        while all(img is None for img in self.imgs):
+            time.sleep(1 / max(self.fps))
+            if not all(x.is_alive() for x in self.threads) or cv2.waitKey(1) == ord('q'):  # q to quit
+                cv2.destroyAllWindows()
+                raise StopIteration
+
+        # Reset self.imgs and return available images
+        self.imgs, im0 = [None] * len(self.imgs), self.imgs
+        available = [i for i, img in enumerate(im0) if img is not None]
```
Comment (glenn-jocher) [inline ultralytics/yolo/data/dataloaders/stream_loaders.py:110]: We can not return anything other than an identically shaped output for every frame. If multiple streams are present and some are lagging then we can return a black frame for these, but understandably every frame in the dataloader needs to be the same shape. The `available` variable here violates this.

**Comment by glenn-jocher** [comment]
@rnoriega updated this with a few changes and merged. Please test main branch and see if this has resolved your issue.

**Pull request merged**